test/index.js

```diff
@@ -1,41 +1,47 @@
-var should = require('should');
+var should = require('should'),
+    superagent = require('superagent');
 
 describe('superagent-use', function() {
-  var superagent;
+  var request;
+
   var prefix = 'http://example.com';
-  var prefixMiddleware = function(request) {
-    if(request.url[0] === '/') {
-      request.url = prefix + request.url;
+  var prefixMiddleware = function(req) {
+    if(req.url[0] === '/') {
+      req.url = prefix + req.url;
     }
-    return request;
+    return req;
   };
 
   beforeEach(function() {
-    /* superagent-use modifies the superagent object, so we need to remove
-       the `require` cache in order to get a new instance for each test. */
-    delete require.cache[require.resolve('superagent')];
-    superagent = require('superagent');
-    require('..')(superagent);
+    request = require('..')(superagent);
```

Comment:
In the tests I want to preserve the `superagent` because I reuse it [here](https://github.com/jonatanklosko/superagent-use/blob/4d3735616d22c6a9662bdde81354626d39795a89/test/index.js#L39-L40). I'll use `agent` instead of `request` as you suggested.